makevsix.js

```diff
@@ -0,0 +1,111 @@
+// 
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+require('shelljs/global');
+var path = require('path'),
+    fs = require('fs');
+var commandLineArgs = require('command-line-args');
+
+var devManifestOverride = {
+    public: false,
+    name: "Cordova Build-Dev",
+    id: "cordova-extension-dev",
+    publisher: "ms-mobiledevops-test"
+}
+
+var prodManifestOverride = {
+    public: true
+}
+
+var cli = commandLineArgs([
+  { name: 'maketest', alias: 't', description: 'make test VSIX', type: Boolean },
+  { name: 'makeprod', alias: 'p', description: 'make production VSIX', type: Boolean },
+  { name: 'publishtest', alias: 'b', description: 'publish test VSIX', type: Boolean },
+  { name: 'skipinstalldeps', alias: 's', description: 'skip npm installing task dependencies (to speed up)', type: Boolean },
+]);
+
+var options;
+try {
+    options = cli.parse()
+}
+catch(err) {
+    echo(err);
+    invalidUsage();
+}
+
+if (Object.keys(options).length === 0)
+    invalidUsage();
+
+if (! exec("npm --version", {silent: true})) {
+    echo("npm not found. please install npm and run again.")
+    exit(1);
+}
+
+if (exec('tfx --version', {silent: true}).code != 0)
+    echoAndExec("tfx-cli not found. installing", "npm install -g tfx-cli");
+
+if (! options.skipinstalldeps)
+    installTasks()
+
+if (options.makeprod)
+    echoAndExec('Creating VSIX', 'tfx extension create --manifest-globs mobiledevopscordovaextension.json --override ' + toOverrideString(prodManifestOverride));
+
+if (options.maketest)
+    echoAndExec('Creating test VSIX',
+        'tfx extension create --manifest-globs mobiledevopscordovaextension.json --override ' + toOverrideString(devManifestOverride));
+
+if (options.publishtest) {
+    var accessToken = env['PUBLISH_ACCESSTOKEN'];
+    if (accessToken === undefined) {
+        echo("Must set PUBLISH_ACCESSTOKEN environment variable to publish a test VSIX");
+        exit(1);
+    }
```

Comment:
This block could actually be simplified furhter wihtout actually having to do this options check. What we could do is always use --override with another var, say overrides, which we set above to be either the dev override or the production override depending on if dev or prod was passed in. For the command, the publish and create take pretty much the same args, so we could just pass that directly the executed line. That simplfies this a lot and actually allows us to add more configurations easily later wihtout having the make new cases here.